### ISSUE TITLE
application closes when opening ssh tunnel

### ISSUE THREAD

**Issue by avalonix** [isontheline/pro.webssh.net#18]
iPad Pro iOS 13.6

**Comment by isontheline** [comment]
Thanks for your report @avalonix 

Would you like to beta test WebSSH Pro in order to send me crash reports?

**Comment by avalonix** [comment]
I'm ready to help you make the app better!  I really love your product and am glad that you continue to support it!  support for ssh tunnels is very important to me.  I am connecting via rdp to my home pc.  there was an annoying flaw in the previous version of your application.  the tunnel worked, but in new versions of ios, the application fell asleep in the background and the connection was interrupted.

**Comment by isontheline** [comment]
@avalonix Thanks for your support !
Yes apps can't do a lot of things in background. 
Does the tunnel works when splitting your screen with RDP app?

**Comment by avalonix** [comment]
I can't check it now.  today updated the application and now the tunnel does not start at all.  the application crashes.  this was the reason to write to you here.

**Comment by avalonix** [comment]
earlier on the old version I didn't think to check two applications on the screen

**Comment by isontheline** [comment]
@avalonix Please use this TestFlight link in order to send me your crash reports :
https://testflight.apple.com/join/QSrBK59z

Thanks

**Comment by nborof** [comment]
I am also experiencing this. It happens whenever the tunnel connects (or directly before). If the tunnel connection is to a server that is unavailable, the app does not crash (it just keeps waiting to find the server). But all other tunnels crash upon connection.

**Comment by isontheline** [comment]
@nborof Thanks for your feedback!
Have you also installed the beta TestFlight binary from https://testflight.apple.com/join/QSrBK59z in order to report your crashes?

**Comment by isontheline** [comment]
Received @avalonix crash report

```
Last Exception Backtrace:
0   CoreFoundation                	0x1b5703344 __exceptionPreprocess + 228 (NSException.m:199)
1   libobjc.A.dylib               	0x1b5418cc0 objc_exception_throw + 60 (objc-exception.mm:565)
2   CoreFoundation                	0x1b5600878 +[NSException raise:format:arguments:] + 100 (NSException.m:146)
3   Foundation                    	0x1b5a52160 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 132 (NSException.m:231)
4   FrontBoardServices            	0x1ba991c60 -[FBSSerialQueue assertOnQueue] + 236 (FBSSerialQueue.m:98)
5   FrontBoardServices            	0x1ba944718 -[FBSSceneImpl updateClientSettings:withTransitionContext:] + 80 (FBSSceneImpl.m:190)
6   FrontBoardServices            	0x1ba94499c -[FBSSceneImpl updateClientSettingsWithTransitionBlock:] + 160 (FBSSceneImpl.m:213)
7   FrontBoardServices            	0x1ba94489c -[FBSSceneImpl updateClientSettingsWithBlock:] + 128 (FBSSceneImpl.m:203)
8   UIKitCore                     	0x1b93843d4 -[FBSScene(UIApp) updateUIClientSettingsWithBlock:] + 184 (FBSScene+UIApp.m:41)
9   UIKitCore                     	0x1b981fe94 -[UIApplication _setIdleTimerDisabled:forReason:] + 224 (UIApplication.m:1783)
10  WebSSH                        	0x1026495d0 -[TUNNELWrapper connectToHost:onPort:withUser:password:andSshKey:] + 1900 (TUNNELWrapper.m:268)
11  WebSSH                        	0x102648d14 -[TUNNELWrapper connectNowWithBlock:] + 84 (TUNNELWrapper.m:72)
12  Foundation                    	0x1b5b08e38 __NSThread__start__ + 852 (NSThread.m:724)
13  libsystem_pthread.dylib       	0x1b5409914 _pthread_start + 168 (pthread.c:896)
14  libsystem_pthread.dylib       	0x1b54119d4 thread_start + 8
```

I am fixing it

**Comment by isontheline** [comment]
Should be fixed by 14.0.3
Waiting for Apple to approve the beta binary

**Comment by nborof** [comment]
@isontheline I installed the testflight and it should have sent the crash info.

**Comment by isontheline** [comment]
@nborof Thanks!
You will notified in a few hours of the availability of WebSSH 14.0.3 (14.0.1 and 14.0.2 don't contain the fix).

**Comment by nborof** [comment]
Just downloaded 14.0.4. The tunnel connects (and works!) without crashing. 

However, when I try to disconnect the tunnel, nothing seems to happen. No matter how much I tap the name of the connection, the dot stays red.  But if I then try to connect via the tunnel, it no longer works. 

So it appears that it does disconnect the tunnel, but the dot stays red, and the only way I can reconnect is to force-quit the app and start again.

**Comment by isontheline** [comment]
@nborof Great!
OK no problem I will fix the red dot asap

**Comment by isontheline** [comment]
Fixed by 14.0.4

I will fix the red dot asap in issue #27
